### DISCUSSION TITLE
user.absent giving error

### DISCUSSION THREAD

**Discussion by bradawk1** [saltstack/salt#67104] [Q&A]
We just moved our salt version up to 3006.7.  For several years we've had code:
`{% for user in ['games','abrt'] %}
{{ usr }}:
  user.absent
{% endfor %}`
and that has worked without a problem.  We are now testing on RHEL 9 and with that code, I am getting:
`Rendering SLS 'my-branch:compliance/harden/init' failed: expected '<document start>', but found '<block mapping start>' in "<unicode string>", line 23, column 1:
   games:
   ^`
Does anyone have an idea why this is happening and how to fix?

**Comment by bdrx312** [comment]
I am surprised that ever worked since that would put multiple states on the same line after the jinja is rendered like this:

```yaml
games: user.absent abrt: user.absent
```

You can run with `-l all` to view the rendered yaml from the jinja.

You just probably just need some newlines:

```
{%- for user in ['games','abrt'] %}
{{ usr }}: 
   user.absent: []
{%- endfor %}
```

I like to put the empty argument list `: []` to make the structure always be a dict to be consistent with my other states, but it should not be necessary.

**Reply by bradawk1** [reply]
I originally used the documentation: [user](https://docs.saltproject.io/en/3006/ref/states/all/salt.states.user.html).  It shows
`testuser:
  user.absent`
with no colon after user.absent.  So, I copied that portion of code to a new file and added the colon after absent.  I then ran salt-call slsutil.renderer salt://compliance/harden/crap.sls saltenv=my-branch and got:
`local:
      ---------
      games:
              ---------
              user.absent:
      abrt:
           ---------
           user.absent:`

I removed the colon and re-ran and got:
`local:
      ---------
      games:
              user.absent
      ---------
      abrt:
              user.absent`

I then ran that state as is and it went fine.  So, it is not this section of code.  Something else in my init.sls must be messing with salt 3006.7 and RHEL 9.  I'll keep looking.
Thanks for taking a look!
Sorry, none of my line feeds in the code sections seem to be working?

**Comment by bradawk1** [comment]
I figured it out.  Two file.absent stanzas at the beginning of the state file were missing trailing colons.  Not sure why that was not caught earlier.   Thanks again!
